client/src/pages/SignUp.tsx

```diff
@@ -26,34 +64,74 @@ const SignUp: React.FC<SignUpProps> = ({ onSignIn }) => {
               <div className="flex flex-col">
                 <div className="flex flex-row pb-8 mt-10">
                   <div className="">
-                    <InputField label="First Name" type="text" />{" "}
+                    <InputField
+                      label="First Name"
+                      type="text"
+                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                        setSignUpInfo({
+                          ...signUpInfo,
+                          firstName: e.target.value,
+                        })
+                      }
+                    />{" "}
                   </div>
                   <div className="pl-5">
-                    <InputField label="Last Name" type="text" />
+                    <InputField
+                      label="Last Name"
+                      type="text"
+                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                        setSignUpInfo({
+                          ...signUpInfo,
+                          lastName: e.target.value,
+                        })
+                      }
+                    />
                   </div>
                 </div>
                 <div className="">
-                  <InputField label="Queen's Email" type="text" />{" "}
+                  <InputField
+                    label="Queen's Email"
+                    type="text"
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                      setSignUpInfo({
+                        ...signUpInfo,
+                        email: e.target.value,
+                      })
+                    }
+                  />{" "}
                 </div>
                 <div className="py-8">
-                  <InputField label="Password" />
+                  <InputField
+                    label="Password"
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                      setPassword1(e.target.value)
+                    }
```

Comment:
Should be password type so password cannot be viewed while user is typing

Reply:
nice catch!